scripts/chgres_cube.py

```diff
@@ -0,0 +1,286 @@
+        config=config_file,
+        cycle=cycle,
+        key_path=key_path,
+    )
+    rundir = Path(chgres_cube_driver.config["rundir"])
+    print(f"Will run in {rundir}")
+
+    # Dereference cycle for file paths
+    expt_config_cp = get_yaml_config(deepcopy(expt_config.data))
+    expt_config_cp.dereference(
+        context={
+            "cycle": cycle,
+            **expt_config_cp,
+        }
+    )
+    chgres_cube_config = _walk_key_path(expt_config_cp, key_path)
+    input_type = chgres_cube_config["chgres_cube"]["namelist"]["update_values"][
+        "config"
+    ].get("input_type")
+
+    # update config for ics task, run and stage data
+    if "task_make_ics" in key_path:
+        varsfilepath = chgres_cube_config["input_files_metadata_path"]
+        shconfig = _parse_var_defns(varsfilepath)
+        extrn_config_fns = shconfig["EXTRN_MDL_FNS"]
+        extrn_config_fhrs = shconfig["EXTRN_MDL_FHRS"]
+
+        if input_type == "grib2":
+            fn_grib2 = extrn_config_fns[0]
+            update = {"grib2_file_input_grid": fn_grib2}
+        else:
+            fn_atm = extrn_config_fns[0]
+            fn_sfc = extrn_config_fns[1]
+            update = {"atm_files_input_grid": fn_atm, "sfc_files_input_grid": fn_sfc}
+        if expt_config["task_get_extrn_ics"]["EXTRN_MDL_NAME_ICS"] in [
+            "HRRR",
+            "RAP",
+        ]:
+            if expt_config["workflow"]["SDF_USES_RUC_LSM"] is True:
+                update["nsoill_out"] = 9
+        else:
+            if expt_config["workflow"]["SDF_USES_THOMPSON_MP"] is True:
+                update["thomp_mp_climo_file"] = expt_config["workflow"][
+                    "THOMPSON_MP_CLIMO_FP"
+
+                ]
+
+        update_cfg = {
+            "task_make_ics": {
+                "chgres_cube": {"namelist": {"update_values": {"config": update}}}
+            }
+        }
+        expt_config_cp.update_from(update_cfg)
+
+        # reinstantiate driver
+        chgres_cube_driver = ChgresCube(
+            config=expt_config_cp,
+            cycle=cycle,
+            key_path=key_path,
+        )
+        chgres_cube_driver.run()
+
+        # Deliver output data to a common location above the rundir.
+        links = {}
+        tile_rgnl = expt_config["constants"]["TILE_RGNL"]
+        nh0 = expt_config["constants"]["NH0"]
+
+        output_dir = os.path.join(rundir.parent, "INPUT")
+        os.makedirs(output_dir, exist_ok=True)
+        links[
+            f"{nco_net}.t{cyc}z{dot_ensmem}.gfs_data.tile{tile_rgnl}.halo{nh0}.nc"
+        ] = str(rundir / f"out.atm.tile{tile_rgnl}.nc")
+        links[
+            f"{nco_net}.t{cyc}z{dot_ensmem}.sfc_data.tile{tile_rgnl}.halo{nh0}.nc"
+        ] = str(rundir / f"out.sfc.tile{tile_rgnl}.nc")
+        links[f"{nco_net}.t{cyc}z.gfs_ctrl.nc"] = str(rundir / f"gfs_ctrl.nc")
+        links[f"{nco_net}.t{cyc}z{dot_ensmem}.gfs_bndy.tile{tile_rgnl}.f000.nc"] = str(
+            rundir / f"gfs.bndy.nc"
+        )
```

Comment:
I started off with the mirrored `ics`/`lbcs` logic, but since this doesn't have any runtime requirements, you're right about moving it out. Would it be worth to move the lbcs outputs to the YAML as well and just insert `t{cyc}z` at runtime?

Reply:
Hrm, testing this and thinking about your comments on slack, its difficult to avoid changing the config at runtime. Well, at least the LBCS files because  `fcst_hhh_FV3LAM` has to be computed

Reply:
In discussion offline, I'll move that to an env variable which will work great